File: 기출/2021kakao/2.js
function solution(needs, r) {
  let answer = 0;
  let info = {};

  if (r === 0) {
    return answer
  }

  for (let i = 0; i < needs.length; i++) {
    let aa = '';
    for (let j = 0; j < needs[i].length; j++) {
      if (needs[i][j] == 1) {
        let a = String(j)
        aa += a
      }
      if (aa.length > r) {
        break
      }
    }
    if (aa.length <= r) {
      if (aa in info) {
        info[aa] += 1
      }
      else {
        info[aa] = 1
      }
    }
  }

  for (let key1 in info) {
    let cnt = 0;
    for (let key2 in info) {
      if (key1.includes(key2)) {
        cnt += info[key2]
      }
    }

    if (cnt > answer) {
      answer = cnt
    }
  }
  console.log(answer);
  return answer;
}

solution([ [ 1, 0, 0 ], [1, 1, 0], [1, 1, 0], [1, 0, 1], [1, 1, 0], [0, 1, 1] ], 2)

File: 기출/2021하반기_한솔/2.py
def solution(s):
    answer = 0
    length = len(s)
    cnt = 1
    li = set()
    while cnt < length + 1:
        for i in range(length):
            if i + cnt > length:
                break
            t = set(s[i:i + cnt])
            if len(t) == cnt:
                li.add(s[i:i + cnt])
        cnt += 1
    answer = len(li)
    print(answer)
    return answer

solution("abac")